### ISSUE TITLE
Failed with exit code 139 (rc=35584),failed to output trimmed reads

### ISSUE THREAD

**Issue by OnlyHigh** [marbl/canu#850]
Good night, 

Trying to assembly ONT data of the Lambda phague, 
I got the following error, 

```
ERROR:  Failed with exit code 139.  (rc=35584)
ERROR:
ABORT:
ABORT: Canu snapshot v1.7 +51 changes (r8743 811868daf236e6238674841a750ad51792129be6)
ABORT: Don't panic, but a mostly harmless error occurred and Canu stopped.
ABORT: Try restarting.  If that doesn't work, ask for help.
ABORT:
ABORT:   failed to output trimmed reads.
ABORT:
ABORT: Disk space available:  93.331 GB
ABORT:
ABORT: Last 50 lines of the relevant log file (./Lambda.trimmedReads.fasta.err):
ABORT:
ABORT:   Dumping reads from 1 to 14382 (inclusive).
ABORT:   
ABORT:   Failed with 'Segmentation fault'; backtrace (libbacktrace):
ABORT:   AS_UTL/AS_UTL_stackTrace.C::97 in _Z17AS_UTL_catchCrashiP9siginfo_tPv()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   AS_UTL/AS_UTL_fasta.C::53 in _Z17AS_UTL_writeFastAP8_IO_FILEPciiS1_z()
ABORT:   stores/gatekeeperDumpFASTQ.C::482 in main()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   (null)::0 in (null)()
ABORT:   Segmentation fault (core dumped)
```
I'm running Canu v1.7 (Today's version) on Ubuntu 16.04, 
With the following line: 

`~/canu/Linux-amd64/bin/canu -trim-assemble -p Lambda -d Lambdassembly3 genomeSize=48k stopOnReadQuality=false overlapper=mhap utgReAlign=true -nanopore-raw Lambda.fastq`

I also tried with the following command line, getting the same error:
`~/canu/Linux-amd64/bin/canu -trim-assemble -p Lambda -d Lambdassembly3 genomeSize=48k overlapper=mhap utgReAlign=true -nanopore-raw Lambda.fastq`

I attach the relevant .log file,
Can someone help me please? 

Thanks in advance, 
Luis Alfonso. 
 
[Lambda.trimmedReads.fasta.err.txt](https://github.com/marbl/canu/files/1866817/Lambda.trimmedReads.fasta.err.txt)

**Comment by skoren** [comment]
You're running with correction turned off on the raw reads, it's likely no overlaps are found in trimming and no reads make it through. Don't specify -trim-assemble, just let it run as default on raw data. I'd also suggest using the 1.7 release not tip as there have been significant changes since the release.

**Comment by OnlyHigh** [comment]
Using the error correction phase, with this command line: 
`~/canu/Linux-amd64/bin/canu -correct -p Lambda -d Lambdacorr2 genomeSize=48k stopOnReadQuality=false maxThreads=7 -nanopore-raw Lambda.fastq`

I move to the trim-assemble: 
`~/canu/Linux-amd64/bin/canu -trim-assemble -p Lambda -d Lambdassembly5 genomeSize=48k stopOnReadQuality=false overlapper=mhap utgReAlign=true maxThreads=7 -nanopore-corrected Lambda.correctedReads.fasta.gz `

It gets hanged in: 

```
-- Running jobs.  First attempt out of 2.
----------------------------------------
-- Starting 'utgmhap' concurrent execution on Mon Apr  2 16:48:01 2018 with 30.056 GB free disk space (1 processes; 1 concurrently)

    cd unitigging/1-overlapper
    ./mhap.sh 1 > ./mhap.000001.out 2>&1
```

**Comment by OnlyHigh** [comment]
After running that command line for the Lambda phague:
`~/canu/Linux-amd64/bin/canu -correct -p Lambda -d Lambacorr2 genomeSize=48k stopOnReadQuality=false maxThreads=7 -nanopore-raw Lambda.fastq`

`~/canu/Linux-amd64/bin/canu -trim-assemble -p Lambda -d Lambdassembly5 genomeSize=48k stopOnReadQuality=false overlapper=mhap utgReAlign=true maxThreads=7 -nanopore-corrected Lambda.correctedReads.fasta.gz `

I got this assemble:

```
>tig00000001 len=78900 reads=7906 covStat=3830.68 gappedBases=no class=contig suggestRepeat=no suggestCircular=no
CGTTGATAAGTCGCAGATCAGCAGGTGGAAGAGGACTGGATTCCAAAGTTCTCAATGCTGCTTGCTGTTCTTGGATGGGGTCGTTGACGACGACATGGCT
CGATTGGCGCGACAAGTTGCTGCGATTCTCACCAATAAAACGCCCGGCGGCAGGAGCGTTCTGAACAAATCAGATGGAGTTCTGAGGTCATTACTGGATC
TATCAACAGGAGTCATTATGACAAATACAGCAAAATACTCAACTTCGGCAGAGGTAGTAACTTTGCCGGACAGGAGCGTAATGTAGCAGATCCTCGATGA 
```

The usual length is 48,502, and I think the CovStat is really high and it's not suggesting circularity. 
I think I might be missconfiguring the overlapper, 
Which commands should I use? 

Thank you!

**Comment by skoren** [comment]
Covstat is the probability of repeat vs unique so this is just indicating it is unique. 

Have you tried checking for self-alignment of the contig to itself or aligning it against the reference? I expect you have very high coverage and so you end up with a lot of long reads making the genome larger than expected. The high coverage/variation in the sample is probably why it's not marked as circular.

**Comment by brianwalenz** [comment]
Can you upload the raw reads?

**Comment by OnlyHigh** [comment]
I sent it through ftp with "albacore-output" name in tar.gz format.
Cheers!

**Comment by OnlyHigh** [comment]
Hey, should I close this issue?

**Comment by brianwalenz** [comment]
I did look at these reads, and remember that sampling down (options readSamplingCoverage and readSamplingBias in the latest code) resulted in the correct assembly.  I forget exactly the parameters I tested with.  If the coverage is too low, the corrected reads aren't as good; too high, and the assembly graph is confused.  Read length adds another dimension that I understand even less - shorter reads _might_ be higher quality, but longer reads are more useful for assembly.
